internal/service/s3/bucket_lifecycle_configuration.go

```diff
@@ -1071,3 +1074,78 @@ func (m transitionDefaultMinimumObjectSizeDefaultModifier) PlanModifyString(ctx
 	}
 	resp.PlanValue = v
 }
+
+var (
+	_ validator.Bool = warnIfSetWithValidator{}
+)
+
+func warnIfSetWith(expressions ...path.Expression) validator.Bool {
+	return warnIfSetWithValidator{
+		PathExpressions: expressions,
+	}
+}
+
+type warnIfSetWithValidator struct {
+	PathExpressions path.Expressions
+}
+
+func (v warnIfSetWithValidator) Description(ctx context.Context) string {
+	return v.MarkdownDescription(ctx)
+}
+
+func (v warnIfSetWithValidator) MarkdownDescription(_ context.Context) string {
+	return fmt.Sprintf("Ensure that if an attribute is set, a warning is emitted if any of these are also set: %q", v.PathExpressions)
+}
+
+// Validation logic is adapted from the standard ConflictsWith validator
+// available for all types
+func (v warnIfSetWithValidator) ValidateBool(ctx context.Context, req validator.BoolRequest, resp *validator.BoolResponse) {
```

Comment:
I've left this alongside the resource definition for now in case further customization is required. Assuming this generic implementation remains, we can migrate this in the future to a shared location (`internal/framework`) and extend it to support all types, similar to how the standard `ConflictsWith` validator works.

Reply:
I think this is a good approach